### PR TITLE
New linter: dls

### LINKED ISSUE #1812: Add support for dls
**Name:** dls
**URL:** https://github.com/LaurentTreguier/dls

A language server for D code.

### PR DESCRIPTION
**Pull request by aurieh**
(Hopefully) resolves #1812 

**Note**: the server doesn't accept change notifications. Initial load doesn't seem to work either. I'll do some debugging and see what happens.

### CODE REVIEW

File: ale_linters/d/dls.vim [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    return ale#Var(a:buffer, 'd_dls_executable')
+endfunction
+
+function! ale_linters#d#dls#GetLanguage(buffer) abort
+    return 'd'
+endfunction
+
+function! ale_linters#d#dls#FindProjectRoot(buffer) abort
+    " Note: this will return . if dub config is empty
+    " dls can run outside DUB projects just fine
```
Comment (w0rp) [inline ale_linters/d/dls.vim]: Thank you for adding the comment. I was going to point this out.

File: ale_linters/d/dls.vim [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+call ale#Set('d_dls_executable', 'dls')
+
+function! ale_linters#d#dls#GetExecutable(buffer) abort
+    return ale#Var(a:buffer, 'd_dls_executable')
+endfunction
+
+function! ale_linters#d#dls#GetLanguage(buffer) abort
+    return 'd'
+endfunction
```
Comment (w0rp) [inline ale_linters/d/dls.vim]: You can remove this function and `language_callback` below. The language given to the `Define` function will be used by default.

Comment (aurieh) [reply ale_linters/d/dls.vim]: Yeah, I'm used to the old way. Will change in a sec.

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**